src/integration-test/java/io/aiven/connect/jdbc/AbstractIT.java

```diff
@@ -0,0 +1,141 @@
+ * limitations under the License.
+ */
+
+package io.aiven.connect.jdbc;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Future;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+import org.apache.avro.generic.GenericRecord;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.junit.jupiter.Container;
+
+public abstract class AbstractIT {
+
+    static final Logger LOGGER = LoggerFactory.getLogger(AbstractIT.class);
+
+    protected static final String TEST_TOPIC_NAME = "test_pg_topic";
+
+    protected static final int OFFSET_FLUSH_INTERVAL_MS = 5000;
+
+    protected static KafkaProducer<String, GenericRecord> producer;
+
+    @Container
+    protected KafkaContainer kafkaContainer =
+            new KafkaContainer().withEnv("KAFKA_AUTO_CREATE_TOPICS_ENABLE", "false");
+
+    @Container
+    protected SchemaRegistryContainer schemaRegistryContainer;
+
+    protected JdbcConnectService jdbcConnectService;
+
+    @BeforeEach
+    void startKafka() throws Exception  {
+        LOGGER.info("Configure Kafka connect plugins");
+        final var pluginDir = setupPluginDir();
+        setupKafka();
+        setupKafkaConnect(pluginDir);
+        createProducer();
+    }
+
+    protected AdminClient adminClient() {
+        final Properties adminClientConfig = new Properties();
+        adminClientConfig.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaContainer.getBootstrapServers());
+        return AdminClient.create(adminClientConfig);
+    }
+
+    private static Path setupPluginDir() throws Exception {
+        final var testDir = Files.createTempDirectory("aiven-kafka-connect-jdbc-test-");
+        final var distFile = Paths.get(System.getProperty("integration-test.distribution.file.path"));
+        assert Files.exists(distFile);
+
+        final var pluginDir = Paths.get(testDir.toString(), "plugins/aiven-kafka-connect-jdbc/");
+        Files.createDirectories(pluginDir);
+
+        final String cmd = String.format("tar -xf %s --strip-components=1 -C %s",
+                distFile.toString(), pluginDir.toString());
+        final Process p = Runtime.getRuntime().exec(cmd);
+        assert p.waitFor() == 0;
+        return pluginDir;
+    }
+
+    private void setupKafka() throws Exception {
+        LOGGER.info("Setup Kafka");
+        kafkaContainer.start();
```

Comment:
I don't remember any need to manually start and stop containers. Testcontainer-JUnit integration should handle containers' life cycle. I guess `@Testcontainers` annotation is needed on the test classes for this.

Reply:
true fixed it